contracts/rewards/emission_manager_library.cairo

```diff
@@ -0,0 +1,214 @@
+        return ()
+    end
+
+    func set_distribution_end{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        asset : felt, reward : felt, new_distribution_end : felt
+    ):
+        assert_only_emission_admin(reward)
+        # TODO
+        # Call reward controllers function.
+        return ()
+    end
+
+    func set_emission_per_second{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        asset : felt,
+        rewards_len : felt,
+        rewards : felt*,
+        new_emissions_per_second_len : felt,
+        new_emissions_per_second : felt*,
+    ):
+        # TODO
+        # Call reward controllers function.
+        _validate_emission_admins(rewards_len, rewards)
+        return ()
+    end
+
+    func set_claimer{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        user : felt, claimer : felt
+    ):
+        Ownable.assert_only_owner()
+
+        let (reward_controller_address) = EmissionManager_rewards_controller.read()
+
+        # TODO
+        # Properly implement set_claimer in reward
+        # IRewardsController.set_claimer(
+        #     contract_address=reward_controller_address, user=user, claimer=claimer
+        # )
+        return ()
+    end
+
+    func set_rewards_controller{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        rewards_controller : felt
+    ):
+        Ownable.assert_only_owner()
+
+        EmissionManager_rewards_controller.write(rewards_controller)
+
+        return ()
+    end
+
+    func set_emission_admin{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        reward : felt, admin : felt
+    ):
+        Ownable.assert_only_owner()
+
+        let (old_admin) = EmissionManager_emission_admins.read(reward)
+        EmissionManager_emission_admins.write(reward, admin)
+
+        emission_admin_updated.emit(old_admin=old_admin, new_admin=admin)
+
+        return ()
+    end
+
+    # Getters
+
+    func get_rewards_controller{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        ) -> (res : felt):
+        let (res) = EmissionManager_rewards_controller.read()
+        return (res)
+    end
+
+    func get_emission_admin{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        reward : felt
+    ) -> (res : felt):
+        let (res) = EmissionManager_emission_admins.read(reward)
+        return (res)
+    end
+
+    # Internals
+
+    func _get_rewards_from_config{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(
+        config_len : felt,
+        config : RewardsDataTypes.RewardsConfigInput*,
+        rewards_len : felt,
+        rewards : felt*,
+    ) -> (
+        config_len : felt,
+        config : RewardsDataTypes.RewardsConfigInput*,
+        rewards_len : felt,
+        rewards : felt*,
+    ):
+        if config_len == 0:
+            return (0, &config[0], rewards_len, rewards)
```

Comment:
is `&config[0]` the same than `config` or I'm missing something?

Reply:
Ha. You're correct, that's late night coding for you.